modules/libcom/src/error/errSymLib.c

```diff
@@ -23,25 +23,25 @@
 #include "cantProceed.h"
 #include "epicsAssert.h"
 #include "epicsStdio.h"
+#include "epicsAtomic.h"
 #include "errMdef.h"
 #include "errSymTbl.h"
 #include "ellLib.h"
 #include "errlog.h"
 
+#define MIN_MODULE_NUM 501
 #define NHASH 256
 
 static epicsUInt16 errhash(long errNum);
 
 typedef struct errnumnode {
-    ELLNODE            node;
     long               errNum;
     struct errnumnode *hashnode;
     const char        *message;
     long               pad;
 } ERRNUMNODE;
```

Comment:
As I understand it up to this point, the removed `node` field and the `hashnode` field have overlapping roles: both make this struct an element of a linked list. Personally, I don't mind a "hand-rolled" linked list, but in that case, the `hashnode` field should be renamed to make it obvious what it represents. On the other hand, it would be preferable to use a consistent API for linked lists everywhere, which means removing the `hashnode` field and restoring `node` instead.

Edit: ok, I get it now. This thing used to be a part of two linked lists, and if you keep `hashnode` instead of `node`, you get to keep existing code for iterating over the hashmap list. I would use this opportunity to change the code, but ultimately, it's up to you. It's just style, after all.

Reply:
The linked list we had before was used only for adding, so I optimized that. The linked list inside the hash table doesn't require a linking to the previous node, therefore the current approach is simpler and with less overhead than using `ELLNODE`